### PR TITLE
node: add tor support

### PR DESCRIPTION
**Pull request by rustaceanrob**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -18,6 +18,13 @@ mod prelude;
 
 use std::net::IpAddr;
 
+#[cfg(feature = "tor")]
+pub use arti_client::TorClient;
+#[cfg(feature = "tor")]
+pub use arti_client::TorClientConfig;
```
Comment (oleonardolima) [inline src/lib.rs:24]: I think you can fit this under a single use, can't you? 🤔

**Review by oleonardolima — COMMENTED** [review]
It's looking good! I'll take a look on pinning the dependencies.

Maybe we should also have a separate example just for Tor usage 🤔

File: src/lib.rs (hunk 1 above)
Comment (rustaceanrob) [reply src/lib.rs:24]: yeah just got lazy

**Comment by rustaceanrob** [comment]
I added Tor to the `memory.rs` example. Could maybe use a rename

**Comment by oleonardolima** [comment]
> I added Tor to the `memory.rs` example. Could maybe use a rename

Yep, I mean was it being used for other purposes, or it's fine to "steal" it for Tor purposes ?

**Comment by rustaceanrob** [comment]
Yeah I think it can demonstrate both. The in-memory peer database should be rare or never used.

**Pull request merged**